components/_patterns/01-atoms/01-links/link/link.twig

```diff
@@ -4,18 +4,18 @@
  * - link_content - the content of the link (typically text)
  * - link_url - the url this link should poing to
  * - link_attributes - array of attribute,value pairs
- * - link_modifiers - array of modifiers to add to the base classname
  * - link_base_class - the base class name
- * - link_blockname - blockname prepended to the base classname
+ * - modifiers - array of modifiers to add to the base classname
+ * - blockname - blockname prepended to the base classname
  *
  * Available blocks:
  * - link_content - used to replace the content of the link
  *     Example: to insert the image component
  */
 #}
-{% set link_base_class_var = link_base_class|default('link') %}
+{% set link_base_class = link_base_class|default('link') %}
 
-<a class="{{ link_base_class_var }}{% for modifier in link_modifiers %} {{ link_base_class_var }}--{{ modifier }}{% endfor %}{% if link_blockname %} {{ link_blockname }}__{{ link_base_class_var }}{% endif %}" {% for attribute,value in link_attributes %}{{ attribute }}="{{ value }}"{% endfor %} href="{{ link_url }}">
+<a{{ bem(link_base_class, (modifiers), blockname) }}{% for attribute,value in link_attributes %} {{ attribute }}="{{ value }}"{% endfor %} href="{{ link_url }}">
```

Comment:
All of these components can span multiple lines because of the way twig handles spaces. It makes it easier to read (classes on one line, attributes on another, href on a third) and tweaks are better represented in diffs since it's not all one line) Here's an example from one of my projects: 
```
{% set link_base_class_var = link_base_class|default('link') %}
<a
  {{ bem(link_base_class_var, (link_modifiers), link_blockname) }}
  {% for attribute,value in link_attributes %}{{ attribute }}="{{ value }}"{% endfor %}
  href="{{ link_url }}"
>
  {% block link_content %}
    {{ link_content }}
  {% endblock %}
</a>
```
Image is another good use-case:
```
<img
  {{ bem(link_base_class_var, (link_modifiers), link_blockname) }}
  src="{{ image_src }}"
  {% if image_alt %}
    alt="{{ image_alt }}"
  {% else %}
    aria-hidden="true"
  {% endif %}
>
```
Link and image are good examples for this since they have multiple distinct parts. If a component only defines classes, I might not do it. e.g. this would look funny and be overkill IMO:
```
<p
  {{ bem(link_base_class_var, (link_modifiers), link_blockname) }}
>
  {% block paragraph_content %}
    {{ paragraph_content }}
  {% endblock %}
</p>
```
I'd just put the bem function inline with the <p> tag and save two lines.